apis/pipelines/hub/pipeline_types.go

```diff
@@ -26,25 +26,76 @@ type PipelineFramework struct {
 	Parameters map[string]*apiextensionsv1.JSON `json:"parameters" yaml:"parameters"`
 }
 
+const DefaultFallbackFramework = "tfx"
+
 func NewPipelineFramework(compilerType string) PipelineFramework {
 	return PipelineFramework{
 		Name:       compilerType,
 		Parameters: make(map[string]*apiextensionsv1.JSON),
 	}
 }
 
+/*
+ComputeHash returns a hash of the pipeline spec, ensuring consistency across versions.
+
+To maintain compatibility, certain fields are deliberately excluded from the hash calculation, or are normalized to match
+the behavior of previous versions when using default fields as determined by conversion to the hub version.
```

Comment:
```suggestion
the behaviour of previous versions when using default fields as determined by conversion to the hub version.
```

Reply:
changed in [a87521b](https://github.com/sky-uk/kfp-operator/pull/643/commits/a87521b1f4ec9d314c91b957edd4800ccce0a6dd)